### PR TITLE
(176) Request a placement - Parole flow

### PR DESCRIPTION
**Pull request by richpjames**
# Context
This adds the third and final flow to the 'Request a placement' form. 
We add 2 screens, one asking about the decision to release and another requesting the user add additional relevant documents.
[Trello card](https:/trello.com/c/ncPgE6LL/176-add-a-flow-to-create-and-submit-a-placement-request)

updates-to-applicatiosn
updates-to-application

### CODE REVIEW

File: server/views/placement-applications/pages/request-a-placement/decision-to-release.njk [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          text: page.questions.informationFromDirectionToRelease,
+          classes: "govuk-label--m"
+        },
+        hint: {
+            html: 
+            'This should include:
+            <ul class="govuk-list govuk-list--bullet govuk-hint">
+                <li>the type of AP placement</li>
+                <li>the placement location</li>
+                <li>any factors that mav affect the release date</li>
```
Comment (pezholio) [inline server/views/placement-applications/pages/request-a-placement/decision-to-release.njk:39]:
```suggestion
                <li>any factors that may affect the release date</li>
```

File: server/form-pages/placement-application/request-a-placement/additionalDocuments.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import type { ApprovedPremisesApplication, Document, PlacementApplication } from '@approved-premises/api'
```
Comment (pezholio) [inline server/form-pages/placement-application/request-a-placement/additionalDocuments.ts:1]: I think this belongs in another commit?

File: server/utils/placementRequests/checkYourAnswersUtils.test.ts [hunk 3]
```diff
@@ -20,47 +21,78 @@ describe('checkYourAnswersUtils', () => {
...
+            ],
           },
-        ])
-      })
+        },
+      ])
     })
   })
+
+  it('if the page name is "additional-documents"', () => {
```
Comment (pezholio) [inline server/utils/placementRequests/checkYourAnswersUtils.test.ts:68]: I feel like this stuff should be done after the additional documents page is added, as it doesn't make sense linerarly otherwise

File: server/form-pages/placement-application/request-a-placement/reasonForPlacement.ts [hunk 4]
```diff
@@ -25,7 +25,17 @@ export default class ReasonForPlacement implements TasklistPage {
...
+      return 'previous-rotl-placement'
+    }
+    if (this.body.reason === 'additional_placement') {
+      return 'additional-placement-details'
+    }
+    if (this.body.reason === 'release_following_decision') {
+      return 'decision-to-release'
+    }
+
+    return ''
```
Comment (pezholio) [inline server/form-pages/placement-application/request-a-placement/reasonForPlacement.ts:38]: Wonder if we should have a case statement and throw an error as the default here? The reason should be one of those three reasons, otherwise we have a problem

Comment (richpjames) [reply server/form-pages/placement-application/request-a-placement/reasonForPlacement.ts:38]: Yeah good point

**Review by pezholio — APPROVED** [review]
LGTM 👍
